### PR TITLE
Issue 16206 fix app config parallel insert

### LINKED ISSUE #16206: Integrity Constraint Violations in server log file after running smashbox test test_uploadFiles.py
### Steps to reproduce

1. Install 8.1 
2. run smashbox test lib/oc-tests/test_uploadFiles
3. Get locking error messages in log file

### Expected behaviour

Test should pass

### Actual behaviour

The test fails with these errors in the server log file:

```

{"reqId":"1vpBZtdhbgFdmwqxXMMN","remoteAddr":"x.x.x.x","app":"webdav","message":"Exception: {\"Message\":\"An exception occurred while executing 'INSERT INTO appconfig (appid, configkey, configvalue) VALUES (?, ?, ?)' with params [\\\"files_sharing\\\", \\\"5\\\", 1431113232.7046]:\\n\\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'files_sharing-5' for key 'PRIMARY'\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(996): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'INSERT INTO app...', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/connection.php(151): Doctrine\\\\DBAL\\\\Connection->executeUpdate('INSERT INTO app...', Array, Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(696): OC\\\\DB\\\\Connection->executeUpdate('INSERT INTO *PR...', Array, Array)\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/appconfig.php(191): Doctrine\\\\DBAL\\\\Connection->insert('*PREFIX*appconf...', Array)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/allconfig.php(148): OC\\\\AppConfig->setValue('files_sharing', '5', 1431113232.7046)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(93): OC\\\\AllConfig->setAppValue('files_sharing', '5', 1431113232.7046)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(107): OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->markDirty(Array, 1431113232.7046)\\n#8 [internal function]: OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->OCA\\\\Files_Sharing\\\\Propagation\\\\{closure}('\\\/localShareDir', Array)\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/hooks\\\/basicemitter.php(99): call_user_func_array(Object(Closure), Array)\\n#10 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/changepropagator.php(80): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\OC\\\\Files', 'propagate', Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/updater.php(92): OC\\\\Files\\\\Cache\\\\ChangePropagator->propagateChanges(NULL)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(184): OC\\\\Files\\\\Cache\\\\Updater->propagate('\\\/localShareDir\\\/...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(98): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #30)\\n#14 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('TEST_FILE_NEW_U...', Resource id #30)\\n#15 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('localShareDir\\\/T...', Resource id #30, NULL)\\n#16 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#17 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#18 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#19 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#20 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(83): Sabre\\\\DAV\\\\Server->exec()\\n#21 \\\/var\\\/www\\\/html\\\/owncloud\\\/remote.php(84): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#22 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":66}","level":4,"time":"2015-05-08T19:27:12+00:00"}
{"reqId":"CWAiH1vtiaP5cnksTlxg","remoteAddr":"x.x.x.x","app":"webdav","message":"Exception: {\"Message\":\"An exception occurred while executing 'INSERT INTO appconfig (appid, configkey, configvalue) VALUES (?, ?, ?)' with params [\\\"files_sharing\\\", \\\"6\\\", 1431113232.8379]:\\n\\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'files_sharing-6' for key 'PRIMARY'\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(996): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'INSERT INTO app...', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/connection.php(151): Doctrine\\\\DBAL\\\\Connection->executeUpdate('INSERT INTO app...', Array, Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(696): OC\\\\DB\\\\Connection->executeUpdate('INSERT INTO *PR...', Array, Array)\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/appconfig.php(191): Doctrine\\\\DBAL\\\\Connection->insert('*PREFIX*appconf...', Array)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/allconfig.php(148): OC\\\\AppConfig->setValue('files_sharing', '6', 1431113232.8379)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(93): OC\\\\AllConfig->setAppValue('files_sharing', '6', 1431113232.8379)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(107): OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->markDirty(Array, 1431113232.8379)\\n#8 [internal function]: OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->OCA\\\\Files_Sharing\\\\Propagation\\\\{closure}('\\\/localShareDir', Array)\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/hooks\\\/basicemitter.php(99): call_user_func_array(Object(Closure), Array)\\n#10 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/changepropagator.php(80): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\OC\\\\Files', 'propagate', Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/updater.php(92): OC\\\\Files\\\\Cache\\\\ChangePropagator->propagateChanges(NULL)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(184): OC\\\\Files\\\\Cache\\\\Updater->propagate('\\\/localShareDir\\\/...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(98): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #30)\\n#14 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('TEST_FILE_NEW_U...', Resource id #30)\\n#15 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('localShareDir\\\/T...', Resource id #30, NULL)\\n#16 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#17 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#18 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#19 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#20 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(83): Sabre\\\\DAV\\\\Server->exec()\\n#21 \\\/var\\\/www\\\/html\\\/owncloud\\\/remote.php(84): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#22 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":66}","level":4,"time":"2015-05-08T19:27:12+00:00"}
{"reqId":"MJ4LZT3lhQhM5OaXJ8xg","remoteAddr":"x.x.x.x","app":"webdav","message":"Exception: {\"Message\":\"An exception occurred while executing 'INSERT INTO appconfig (appid, configkey, configvalue) VALUES (?, ?, ?)' with params [\\\"files_sharing\\\", \\\"7\\\", 1431113232.904]:\\n\\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'files_sharing-7' for key 'PRIMARY'\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/DBALException.php(116): Doctrine\\\\DBAL\\\\Driver\\\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException))\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(996): Doctrine\\\\DBAL\\\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOMySql\\\\Driver), Object(Doctrine\\\\DBAL\\\\Driver\\\\PDOException), 'INSERT INTO app...', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/db\\\/connection.php(151): Doctrine\\\\DBAL\\\\Connection->executeUpdate('INSERT INTO app...', Array, Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Connection.php(696): OC\\\\DB\\\\Connection->executeUpdate('INSERT INTO *PR...', Array, Array)\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/appconfig.php(191): Doctrine\\\\DBAL\\\\Connection->insert('*PREFIX*appconf...', Array)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/allconfig.php(148): OC\\\\AppConfig->setValue('files_sharing', '7', 1431113232.904)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(93): OC\\\\AllConfig->setAppValue('files_sharing', '7', 1431113232.904)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/propagation\\\/recipientpropagator.php(107): OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->markDirty(Array, 1431113232.904)\\n#8 [internal function]: OCA\\\\Files_Sharing\\\\Propagation\\\\RecipientPropagator->OCA\\\\Files_Sharing\\\\Propagation\\\\{closure}('\\\/localShareDir', Array)\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/hooks\\\/basicemitter.php(99): call_user_func_array(Object(Closure), Array)\\n#10 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/changepropagator.php(80): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\OC\\\\Files', 'propagate', Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/updater.php(92): OC\\\\Files\\\\Cache\\\\ChangePropagator->propagateChanges(NULL)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(184): OC\\\\Files\\\\Cache\\\\Updater->propagate('\\\/localShareDir\\\/...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(98): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #30)\\n#14 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('TEST_FILE_NEW_U...', Resource id #30)\\n#15 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('localShareDir\\\/T...', Resource id #30, NULL)\\n#16 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#17 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#18 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#19 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#20 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(83): Sabre\\\\DAV\\\\Server->exec()\\n#21 \\\/var\\\/www\\\/html\\\/owncloud\\\/remote.php(84): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#22 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/doctrine\\\/dbal\\\/lib\\\/Doctrine\\\/DBAL\\\/Driver\\\/AbstractMySQLDriver.php\",\"Line\":66}","level":4,"time":"2015-05-08T19:27:12+00:00"}

```

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration:

ownCloud Enterprise Edition 8.1 beta 1 (testing)

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #16206 

@DeepDiver1975 @LukasReschke @MorrisJobke @jnfrmarks

### CODE REVIEW

File: lib/private/appconfig.php [hunk 1]
```diff
@@ -42,13 +42,14 @@
 
 namespace OC;
 
-use \OC\DB\Connection;
+use OC\DB\Connection;
```
Comment (MorrisJobke) [inline lib/private/appconfig.php]: Why not use `\OCP\IDBConnection`? Then we don't have to care about internal compatibility?

**Comment by karlitschek** [comment]
very  good :-) :+1:

**Comment by DeepDiver1975** [comment]
@nickvergessen did you execute smashbox test_uploadFiles.py on this change? Or in other words - did you see the error as reported in #16206 locally before this change?

@jnfrmarks please test - THX

File: lib/private/appconfig.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/appconfig.php]: Because `\OCP\IConnection::update` does not exist... :(

**Comment by nickvergessen** [comment]
@DeepDiver1975 no, but I wrote a unit test `testSettingConfigParallel()` that fails without this patch, rebuilding the situation, and passes with the fix: https://github.com/owncloud/core/commit/39497b9c3a02c383b2bd660c9114e65b732848e3

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 no, but I wrote a unit test testSettingConfigParallel() that fails without this patch, rebuilding the situation, and passes with the fix: 39497b9

nice job - didn't realize this - THX

:+1:

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**